2016/proposals/chris-medrela.md

```diff
@@ -0,0 +1,560 @@
+
+    - Note that Admins and Head of Training are not engaged at all.
+
+- Amy generates PDF certificates. Those certificates are hosted as static files on Amy (much easier than integrating Amy with github).
+
+- New instructors send their biography as well as photo in Amy; and are asked to join `discuss` as well as `instructors` list. After sending it, they can download their certificate from Amy.
+
+## Milestone 2: Managing Recruiting Candidates
+
+That is managing everything before and during an Instructor Training -- Recruiting Candidates, Matching them to Instructor Trainings and running those trainings.
+
+All Candidates will be represented in `Person` table, from the beginning of the workflow. For simplicity, trainees can login using their emails, so they don't have to choose username. Every Amy user can login using their email address or username (if they have an username).
+
+The new workflow is:
+
+- A Candidate or a Lead Candidate of a group of Candidates uses Registration Form to create his account.
+
+- The Candidate logs into Amy and uses Application Form to apply for an Instructor Training. This form is used for both individual Candidates as well as groups of Candidates. In the case of group applications, the Lead Candidate needs to enter names and email addresses of other Candidates. After submitting the form, a new account is created for each Candidate, but no password is set yet. For simplicity, UI allows the Candidate to request only one Instructor Training.
+
+- The Lead Candidate can later log into Amy and edit his application.
+
+- Admins can view all applications along with candidates in Amy. They can also merge groups and individuals to form bigger groups. They manually match Candidates to Instructor Trainings.
+
+- Admins negotiate date and place of Instructor Training with Lead Candidate via email.
+
+- In the end of negotiation of Instructor Training date, Admins send emails to all Candidates to confirm they want to participate in the Instructor Training.
+
+- An Instructor Training take place.
+
+- At Instructor Training (or after that training), all Trainees already have accounts in Amy. They can set up their passwords using Registration Form.
+
+- After logging in, every Trainee can see the progress of Checkout Process, as described in Milestone 1.
+
+## Schema Changes
+
+```python
+class SWCOrDCField(models.CharField):
+    max_length = STR_MED
+    choices = (
+        ('swc', 'Software-Carpentry'),
+        ('dc', 'Data-Carpentry'),
+    )
+    blank = False
+    default = 'swc'
+
+class Person(AbstractBaseUser, PermissionsMixin):
+    # determines whether the person has access to AMY interface for admins and trainers
+    is_admin = models.BooleanField(default=True)  
+
+    # username -- no longer required
+    # email -- required
+
+    swc_instructor_training = models.ForeignKey('InstructorTraining', null=True)
```

Comment:
I also don't like this kind of duplication, but I don't have any idea how to make it simpler.